atspi-common/src/operation.rs

```diff
@@ -0,0 +1,33 @@
+/// An operation can either be [`Self::Insert`] or [`Self::Delete`].
+/// These correspond to methods available on [`Vec`].
+#[derive(Debug, PartialEq, Clone, serde::Serialize, serde::Deserialize, Eq, Hash, Default)]
+pub enum Operation {
+	#[default]
+	#[serde(rename = "add")]
+	#[serde(alias = "add/system")]
+	Insert,
+	#[serde(rename = "delete")]
+	#[serde(alias = "delete/system")]
+	Delete,
+}
+
+impl TryFrom<&str> for Operation {
+	type Error = crate::AtspiError;
+	fn try_from(s: &str) -> Result<Operation, Self::Error> {
+		match s {
+			"add" | "add/system" => Ok(Operation::Insert),
+			"delete" | "delete/system" => Ok(Operation::Delete),
+			_ => Err(crate::AtspiError::KindMatch(format!("\"{s}\" is not a type of Operation"))),
+		}
+	}
+}
+
+impl Into<String> for Operation {
```

Comment:
About `Into`: 
It is common practice to not implement `Into` because it comes for free with the `From` implementation.
```Rust
impl From<Operation> for String
```
(In argument position however, you generally do want to use `Into` to catch both `From` and `Into` even if `Into` is implemented explicitly.) 

If possible I'd suggest implementing `Display` for `Operation` instead because it is more versatile. But if you have a specific use-case for the conversion, you could consider implementing `From<Operation> for String` instead?